File: VICSApp/CoreSystem/DHCD/FormUI/Meeting/AuthorizationUpdate.cs
using System;
using System.Data;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace pmDHCD
{
    public partial class AuthorizationUpdate
    {
        private int updatedelegate;
        private string updateholdercode;
        private int initdelegateright;

        public AuthorizationUpdate(int delegatecode, string holdercode, int delegateright)
        {
            InitializeComponent();
            updatedelegate = delegatecode;
            updateholdercode = holdercode;
            initdelegateright = delegateright;
            _Button2.Name = "Button2";
            _Button1.Name = "Button1";
        }

        private void AuthorizationUpdate_Load(object sender, EventArgs e)
        {
            DataTable daibieu;
            DataTable codong;
            try
            {
                daibieu = My.MyProject.Forms.Mainform.BenlyDal.Delegate_getlist(My.MyProject.Forms.Mainform.workingmeeting, updatedelegate, "");
                codong = My.MyProject.Forms.Mainform.BenlyDal.Holder_getlist(My.MyProject.Forms.Mainform.workingmeeting, updateholdercode, "");
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi : " + ex.Message);
                return;
            }

            MaskedTextBox2.Text = Conversions.ToString(daibieu.Rows[0]["Delegatecode"]);
            MaskedTextBox3.Text = Conversions.ToString(daibieu.Rows[0]["IdentityCard"]);
            MaskedTextBox4.Text = Conversions.ToString(daibieu.Rows[0]["DelegateName"]);
            MaskedTextBox8.Text = Conversions.ToString(codong.Rows[0]["Holdercode"]);
            MaskedTextBox7.Text = Conversions.ToString(codong.Rows[0]["HolderIdentity"]);
            MaskedTextBox6.Text = Conversions.ToString(codong.Rows[0]["Holdername"]);
            StockTextBox1.Text = Conversions.ToString(codong.Rows[0]["Voterights"]);
            StockTextBox2.Text = initdelegateright.ToString();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            try
            {
                My.MyProject.Forms.Mainform.BenlyDal.Authorizations_update(My.MyProject.Forms.Mainform.workingmeeting, MaskedTextBox8.Text, Conversions.ToDecimal(MaskedTextBox2.Text), Conversions.ToDecimal(StockTextBox2.Text));
                Close();
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi" + ex.Message);
            }
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VICSApp/CoreSystem/DHCD/FormUI/Meeting/AuthorizationUpdate.vb
Public Class AuthorizationUpdate
    Dim updatedelegate As Integer
    Dim updateholdercode As String
    Dim initdelegateright As Integer
    Public Sub New(ByVal delegatecode As Integer, ByVal holdercode As String, ByVal delegateright As Integer)
        InitializeComponent()
        Me.updatedelegate = delegatecode
        Me.updateholdercode = holdercode
        Me.initdelegateright = delegateright
    End Sub
    Private Sub AuthorizationUpdate_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim daibieu As DataTable
        Dim codong As DataTable
        Try
            daibieu = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, updatedelegate, "")
            codong = Mainform.BenlyDal.Holder_getlist(Mainform.workingmeeting, updateholdercode, "")
        Catch ex As Exception
            MsgBox("Lỗi : " + ex.Message)
            Exit Sub
        End Try
        MaskedTextBox2.Text = daibieu.Rows(0).Item("Delegatecode")
        MaskedTextBox3.Text = daibieu.Rows(0).Item("IdentityCard")
        MaskedTextBox4.Text = daibieu.Rows(0).Item("DelegateName")
        MaskedTextBox8.Text = codong.Rows(0).Item("Holdercode")
        MaskedTextBox7.Text = codong.Rows(0).Item("HolderIdentity")
        MaskedTextBox6.Text = codong.Rows(0).Item("Holdername")
        StockTextBox1.Text = codong.Rows(0).Item("Voterights")
        StockTextBox2.Text = initdelegateright

    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        Try
            Mainform.BenlyDal.Authorizations_update(Mainform.workingmeeting, MaskedTextBox8.Text, MaskedTextBox2.Text, StockTextBox2.Text)
            Me.Close()
        Catch ex As Exception
            MsgBox("Lỗi" + ex.Message)
        End Try
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        Me.Close()
    End Sub
End Class